### ISSUE TITLE
Finishing a replay = Timer breaks until game restart

### ISSUE THREAD

**Issue by FEFELAND** [ReadieFur/BSDataPuller#13]
Game Version: 1.16.4
DataPuller Version: 2.0.5

Bug related to replays.
If you finish watching a replay, the time data keeps displaying the length of the song that was replayed while trying to show the current song time as well. This doesn't happen if you pause and exit before the replay is done I think.

Work around seems to be restarting game.

https://user-images.githubusercontent.com/57236015/128587401-ce56dded-2a24-4126-b4a8-4a8f8eb4d016.mp4

**Comment by kOFReadie** [comment]
Hmm, this is odd, weird issue. I'll defo take a look into it later today. I also seemed to experience some very slow data updates in replay mode as well so I'll look into that, although I'm hoping that was just a one time thing.

**Comment by UnskilledFreak** [comment]
Sorry for the delay (months? D:).
I havent looked at this repo so feel free to create an issue on my overlay github page if you encounter any bugs or weird behaviour.
This was a bug many months ago on my overlay where the time calculation breaks if a replay was watched. Game restart fixed it temporarily because the overlay lost connection to the mod, resets itself (drops all buffers and temp stuff) and reconnects to the game again like nothing ever happend.
Long story short; this is fixed :3

**Comment by ReadieFur** [comment]
> Sorry for the delay (months? D:). I havent looked at this repo so feel free to create an issue on my overlay github page if you encounter any bugs or weird behaviour. This was a bug many months ago on my overlay where the time calculation breaks if a replay was watched. Game restart fixed it temporarily because the overlay lost connection to the mod, resets itself (drops all buffers and temp stuff) and reconnects to the game again like nothing ever happend. Long story short; this is fixed :3

It's fixed by itself? (i.e. in a recent game update, or does youre repo have a fix?)

**Comment by UnskilledFreak** [comment]
I included a fix on my overlay in the last releases a few months back. It was a bug on the overlay (client) side, not your mod as far as I know.
I calculated the time in a "smooth" fashion before so the time will update even if no update gets received from the mod but that turns out to be a bad idea. I still dont know why only replay mode tiggered this bug but I rewrote the time section anyways and havent encountered this bug anymore since.
